### PR TITLE
feat(ingestion): make sure all caches are LRU caches and not unbounded maps

### PR DESCRIPTION
**Pull request by yakkomajuri**
Follow up to #12841.

Haven't looked into if there's tests that cover this yet, putting this up so I don't forget that I set out to do this.

### CODE REVIEW

File: plugin-server/src/worker/ingestion/team-manager.ts [hunk 1]
```diff
@@ -78,7 +78,9 @@ export class TeamManager {
         const timeout = timeoutGuard(`Still running "fetchTeam". Timeout warning after 30 sec!`)
         try {
             const team: Team | null = (await this.db.fetchTeam(teamId)) || null
-            this.teamCache.set(teamId, team)
+            if (team) {
```
Comment (macobo) [inline plugin-server/src/worker/ingestion/team-manager.ts:81]: Is this a good idea? This means that we do a team lookup if team has been deleted 100% of the time.

File: plugin-server/src/worker/ingestion/organization-manager.ts [hunk 2]
```diff
@@ -1,44 +1,58 @@
...
+        const cachedOrganization = this.organizationCache.get(organizationId)
         if (cachedOrganization) {
             return cachedOrganization
         }
 
         const timeout = timeoutGuard(`Still running "fetchOrganization". Timeout warning after 30 sec!`)
         try {
             const organization: RawOrganization | null = (await this.db.fetchOrganization(organizationId)) || null
-            this.organizationCache.set(organizationId, [organization, Date.now()])
+            if (organization) {
```
Comment (macobo) [inline plugin-server/src/worker/ingestion/organization-manager.ts:45]: Ditto: I think caching nulls is desireable here.

File: plugin-server/src/worker/ingestion/group-type-manager.ts [hunk 3]
```diff
@@ -1,33 +1,41 @@
...
     constructor(db: DB, teamManager: TeamManager, instanceSiteUrl?: string | null) {
         this.db = db
         this.teamManager = teamManager
-        this.groupTypesCache = new Map()
+        this.groupTypesCache = new LRU({
+            max: 10_000,
+            // 30 seconds
+            maxAge: 30_000,
+            // being explicit about the fact that we want to update
+            // the team cache every 2min, irrespective of the last access
```
Comment (macobo) [inline plugin-server/src/worker/ingestion/group-type-manager.ts:23]: Copy-pasta.

File: plugin-server/src/worker/ingestion/organization-manager.ts [hunk 4]
```diff
@@ -1,44 +1,58 @@
...
         this.db = db
         this.teamManager = teamManager
-        this.organizationCache = new Map()
-        this.availableFeaturesCache = new Map()
+        this.organizationCache = new LRU({
+            max: 10_000,
+            // 30 seconds
+            maxAge: 30_000,
+            // being explicit about the fact that we want to update
+            // the team cache every 2min, irrespective of the last access
```
Comment (macobo) [inline plugin-server/src/worker/ingestion/organization-manager.ts:24]: Copy-pasta

**Comment by yakkomajuri** [comment]
good point about caching nulls, hadn't considered

**Pull request closed without merging**